allauth/account/adapter.py

```diff
@@ -676,6 +679,16 @@ def get_login_stages(self):
             ret.append("allauth.mfa.stages.AuthenticateStage")
         return ret
 
+    def send_notification_mail(self, template_prefix, user, context):
+        if app_settings.ACCOUNT_EMAIL_NOTIFICATIONS:
+            context |= {
+                "timestamp": datetime.now(),
+                "ip": self.get_client_ip(self.request),
+                "browser_agent": self.get_browser_user_agent(self.request),
+            }
+            print(context)
+            self.send_mail(template_prefix, user.email, context)
```

Comment:
`user.email` is not guaranteed to be present. You need to lookup the primary `EmailAddress` of the user.

Reply:
Should we display some warning, if ACCOUNT_EMAIL_NOTIFICATIONS is True and No Email is present?

Reply:
I don't think so, at least, I wouldn't know how that warning could be formulated. "We wanted to send you a mail about changes to your account but we can't because there is no email address attached"? I propose to just silently skip this..

Reply:
We can display one warning in console for that? 
- When user is not using email auth 
-  When user don't have an active email attached.
This will helpful in debugging?

Reply:
You could add a `logging.debug()` statement for this, but I wouldn't want to go much further than that.

Reply:
Noted I'll do